py/server/tests/test_vectorization.py

```diff
@@ -199,6 +201,48 @@ def pyfunc_bool(p1, p2, p3) -> bool:
         self.assertEqual(t1.size, t.size)
         self.assertEqual(9, t.size)
 
+    def test_return_types(self):
+        def pyfunc_bool() -> bool:
+            return True
+
+        def pyfunc_byte() -> np.int8:
+            return 100
+
+        def pyfunc_short() -> np.int16:
+            return 100
+
+        def pyfunc_int() -> np.int32:
+            return 100
+
+        def pyfunc_long() -> np.int64:
+            return 100
+
+        def pyfunc_float() -> np.float32:
+            return 100
+
+        def pyfunc_double() -> np.float64:
+            return 1000
+
+        def pyfunc_str() -> str:
+            return "abc"
+
+        t = empty_table(1).update("X = pyfunc_bool()")
+        self.assertEqual(t.columns[0].data_type, dtypes.bool_)
+        t = empty_table(1).update("X = pyfunc_byte()")
+        self.assertEqual(t.columns[0].data_type, dtypes.byte)
+        t = empty_table(1).update("X = pyfunc_short()")
+        self.assertEqual(t.columns[0].data_type, dtypes.short)
+        t = empty_table(1).update("X = pyfunc_int()")
+        self.assertEqual(t.columns[0].data_type, dtypes.int32)
+        t = empty_table(1).update("X = pyfunc_long()")
+        self.assertEqual(t.columns[0].data_type, dtypes.long)
+        t = empty_table(1).update("X = pyfunc_float()")
+        self.assertEqual(t.columns[0].data_type, dtypes.float32)
+        t = empty_table(1).update("X = pyfunc_double()")
+        self.assertEqual(t.columns[0].data_type, dtypes.double)
+        t = empty_table(1).update("X = pyfunc_str()")
+        self.assertEqual(t.columns[0].data_type, dtypes.JObject)
```

Comment:
dtypes.string is actually mapped to np.object and the reason is that numpy's U type data are not a single format, it could be U1, U2, U3, etc. Right now we rely on JPY to properly convert between Python str and Java String but we haven't had any work done on numpy unicode arrays.